src/pages/en/guides/troubleshooting.md

```diff
@@ -0,0 +1,179 @@
+---
+layout: ~/layouts/MainLayout.astro
+title: Troubleshooting
+description: Need help? Stuck on something? We've got you covered.
+i18nReady: true
+---
+
+Astro provides several different tools to help you troubleshoot and debug your code.
+
+## Common Error Messages
+
+Here are some common error messages you might see in the terminal, what they might mean, and what to do about them.
+
+### Transform failed with X error
+
+This message often appears due to a current limitation in Astro requiring your import and export statements to be at the top of your `.astro` file.
+
+**Solution**: Write your imports and exports at the top of your component script.
+
+**Status**: Current limitation; fix is being worked on.
+
+💡 Not sure that this is your problem? Check to see if anyone else has reported [this issue](https://github.com/withastro/astro/issues?q=is%3Aissue+is%3Aopen+Transform+failed+with+*+error)!
+
+### Cannot use import statement outside a module
+
+In Astro components, `<script>` tags are hoisted and loaded as [JS modules](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Modules) by default. If you have included the [`is:inline` directive](/en/reference/directives-reference/#isinline) or any other attribute in your tag, this default behavior is removed.
+
+**Solution**: If you have added any attributes to your `<script>` tag, you must also add the `type="module"` attribute to be able to use import statements.
+
+**Status**: Expected Astro behavior, as intended.
+
+💡 Not sure that this is your problem? Check to see if anyone else has reported [this issue](https://github.com/withastro/astro/issues?q=is%3Aissue+is%3Aopen+Cannot+use+import+statement)!
+
+### Unable to render component
+
+This is indicates an error in a component you have imported and are using in your Astro template.
+
+**Solution**: Check the appropriate documentation for your [Astro](/en/core-concepts/astro-components/) or [UI framework](/en/core-concepts/framework-components/) component. Consider opening an Astro starter template from [astro.new](https://astro.new) and troubleshooting just your component in a minimal Astro project.
+
+> This can also be caused by attempting to access the `window` or `document` object at render time.
+
+By default, Astro will render your component [isomorphically](https://en.wikipedia.org/wiki/Isomorphic_JavaScript), meaning it runs on the server where the browser runtime is not available. You can disable this pre-render step using [the `client:only` directive](/en/reference/directives-reference/#clientonly).
+
+**Solution**: Try to access those objects after rendering (ex: [`useEffect()`](https://reactjs.org/docs/hooks-reference.html#useeffect) in React or [`onMounted()`](https://vuejs.org/api/composition-api-lifecycle.html#onmounted) in Vue and Svelte)
+
+**Status**: Expected Astro behavior, as intended.
```

Comment:
```suggestion
### Unable to render component

This is indicates an error in a component you have imported and are using in your Astro template.

#### Common cause

This can be caused by attempting to access the `window` or `document` object at render time. By default, Astro will render your component [isomorphically](https://en.wikipedia.org/wiki/Isomorphic_JavaScript), meaning it runs on the server where the browser runtime is not available. You can disable this pre-render step using [the `client:only` directive](/en/reference/directives-reference/#clientonly).

**Solution**: Try to access those objects after rendering (ex: [`useEffect()`](https://reactjs.org/docs/hooks-reference.html#useeffect) in React or [`onMounted()`](https://vuejs.org/api/composition-api-lifecycle.html#onmounted) in Vue and Svelte)

**Status**: Expected Astro behavior, as intended.

#### Not that?

**Solution**: Check the appropriate documentation for your [Astro](/en/core-concepts/astro-components/) or [UI framework](/en/core-concepts/framework-components/) component. Consider opening an Astro starter template from [astro.new](https://astro.new) and troubleshooting just your component in a minimal Astro project.
```